src/index.js

```diff
@@ -1,68 +1,72 @@
 var stores = require('./stores')
-var adaptors = require('./adaptors')
-var fs = require('fs')
-var ncp = require('ncp')
-var path = require('path')
+var extend = require('xtend')
+var fs = require('fs-blob-store')
 
-/**
- * Constructor
- *
- * @param {String} root_path
- * @param {Maybe<Object>} options
- */
-function Repo (root_path, options) {
-  this.root_path = root_path
-  this.options = options || {}
-  this.loaded = false
-}
-
-Repo.prototype = {
-  _chooseAdaptor: function () {
-    var adaptor = adaptors[this.options.adaptor || 'fs-repo']
+exports = module.exports = Repo
 
-    if (!adaptor) {
-      throw new Error('Adaptor "' + this.options.adaptor + '" not supported')
+function Repo (repoPath, options) {
+  var self = this
+  var base = {
+    stores: {
+      keys: fs,
+      config: fs,
+      datastore: fs,
+      logs: fs,
+      locks: fs,
+      version: fs
     }
+  }
+  options = extend(base, options)
 
-    return adaptor
-  },
-
-  exists: function () {
+  self.exists = function () {
     try {
-      return !!fs.statSync(this.root_path)
+      return !!fs.statSync(this.repoPath)
     } catch (err) {
       return false
     }
-  },
+  }
 
-  init: function (config, callback) {
+  self.init = function (config, callback) {
```

Comment:
ahah eventually we will run perf analysis everywhere and there will be a cascade of .prototype and named functions, until then, this makes the code so easy to read :D